### ISSUE TITLE
sonarqube-scanner-build-wrapper-linux is failing "with undefined reference to `main'"

### ISSUE THREAD

**Issue by axel3rd** [SonarSource/sonar-scanning-examples#87]
Recently (~2 months) with some Ubuntu updates, the current `build.sh` could fail with: 

```
$ ./build.sh
REMOVE PREVIOUS BUILD
BUILDING
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/Scrt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
collect2: error: ld returned 1 exit status
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/Scrt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
collect2: error: ld returned 1 exit status
BUILD FAILURE
```

So the `bw-outputs/build-wrapper-dump.json` produced file could be invalid for *sonar-scanner* analysis.

Current environment version:

```
$ g++ --version
g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
```

I'm not export in C++ compilation, but perhaps `-c` flag seems required here:

https://github.com/SonarSource/sonar-scanning-examples/blob/638834d3efb4c97edd3fb440bf60d529e1be2d4b/sonarqube-scanner-build-wrapper-linux/build.sh#L7-L9

**Comment by axel3rd** [comment]
Fix proposal: #88

**Comment by sylvain-combe-sonarsource** [comment]
Hello @axel3rd 
thanks for reporting this issue AND proposing a fix!
